packages/pluggableWidgets/datagrid-web/src/components/ColumnSelector.tsx

```diff
@@ -33,43 +63,73 @@ export function ColumnSelector(props: ColumnSelectorProps): ReactElement {
             {props.columns.map((column: ColumnProperty, index: number) => {
                 const isVisible = !props.hiddenColumns.includes(column.id);
                 return column.canHide ? (
-                    <li key={index}>
+                    <li
+                        key={index}
+                        onClick={e => {
+                            e.preventDefault();
+                            e.stopPropagation();
+                            onClick(isVisible, column.id);
+                        }}
+                        onKeyDown={e => {
+                            if (e.key === "Enter" || e.key === " ") {
+                                e.preventDefault();
+                                e.stopPropagation();
+                                onClick(isVisible, column.id);
+                            } else if (
+                                (e.key === "Tab" &&
+                                    (index === lastHidableColumnIndex ||
+                                        (e.shiftKey && index === firstHidableColumnIndex))) ||
+                                e.key === "Escape"
+                            ) {
+                                e.preventDefault();
+                                setShow(false);
+                                buttonRef.current?.focus();
+                            }
+                        }}
+                        role="menuitem"
+                        tabIndex={0}
+                    >
                         <input
                             checked={isVisible}
                             disabled={isVisible && props.columns.length - props.hiddenColumns.length === 1}
-                            id={`checkbox_toggle_${index}`}
-                            onClick={() => {
-                                props.setHiddenColumns(prev => {
-                                    if (!isVisible) {
-                                        prev.splice(
-                                            prev.findIndex(v => v === column.id),
-                                            1
-                                        );
-                                        return [...prev];
-                                    } else {
-                                        return [...prev, column.id];
-                                    }
-                                });
-                            }}
+                            id={`${props.name}_checkbox_toggle_${index}`}
                             type="checkbox"
+                            tabIndex={-1}
```

Comment:
so we basically we can never target `input`, is that fine?

Reply:
yes, because the LI is the clickable element